keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1129 @@
+* Validate scheduling and admission.
+* Validate the containers with no limits set are throttled on CPU when CPU usage reaches Pod level CPU limits.
+* Validate the containers with no limits set are OOMKilled when memory usage reaches Pod level memory limits.
+
+
+### Graduation Criteria
+
+
+#### Phase 1: Alpha (target 1.31)
+
+
+* Feature is disabled by default. It is an opt-in feature which can be enabled by enabling the `PodLevelResources`
+feature gate and by setting the new `resources` fields in PodSpec at Pod level.
+* Support the basic functionality for scheduler to consider pod-level resource requests to find a suitable node.
+* Support the basic functionality for kubelet to translate pod-level requests/limits to pod-level cgroup settings.
+* Unit test coverage.
+* E2E tests.
+* Documentation mentioning high level design.
+
+
+#### Phase 2:  Beta (target 1.32)
+
+
+* User Feedback.
+* Feature is disabled by default. It is an opt-in feature which can be enabled by setting the new fields in PodSpec.
+* Support for all features that are scoped out of alpha phase in the KEP design.
+* Unit test coverage for additional features supported in Beta.
+* E2E tests for additional features supported in Beta.
+* Documentation update and blog post to announce feature in beta phase.
+* [Tentative] Benchmark tests for resource usage with and without pod level resources for bursty workloads.
+  * Use kube_pod_container_resource_usage metric to check resource utilization.
+
+
+#### GA (stable)
+
+
+* TBD
+
+
+### Upgrade / Downgrade Strategy
+
+The existing workloads will not have any impact specifically because of pod-level resources feature
+since they won't be using the new field `resources` in PodSpec at pod-level. However, if pods 
+that do utilize this feature have poorly configured limits, they could become noisy neighbours and
+negatively impact other workloads on the same node. It's important to nose that this issue of noisy
+neighbours due to incorrect limits it not exclusive to pod-level resources; it can also occur
+at with container-level limits.
+
+If a cluster is upgraded, the cluster administrator will experience pre-upgrade behavior until 
+`resources` field is specified at pod-level. 
+
+If the workloads are not using the new feature, downgrade won't be affected. However, if it is required to
+downgrade a cluster that has workloads using the pod-level limits functionality, following steps need to be
+performed:
+
+* Translate pod-level resource specifications to container-level specifications.
+
+* Delete existing workloads and recreate them with container-level specifications.
+
+* Drain the nodes one by one to prevent the new pods from being scheduled on them.
+
+* Downgrade the kubelet on the drained nodes to a version that doesn't support pod-level resource specification
+fields.
+
+* Downgrade the control plane components.
+
+* Uncordon the nodes, making it available for scheduling again.
+
+
+### Version Skew Strategy
```

Comment:
I would add: if the feature gate is disabled, but the pod-level resources field is set, the Kubelet should reject the pod.

It won't matter much for the alpha case, but it will help with the alpha->beta transition.

Alternatively, we could say that the kubelet always honors the field (basically ignore the feature gate), and only control the feature at the cluster level.